### PR TITLE
adding BoundingBox.extent and RegularMesh.bounding_box

### PR DESCRIPTION
**Pull request by shimwell**
I'm plotting a few regular meshes at the moment and would find it useful to be able to easily get the extent of the regular mesh. 

Adding a ```bounding_box``` property to the ```RegularMesh``` and adding the ```extent``` property to the ```BoundingBox``` would make it easier when plotting.

example usage.

```python
plt.imshow(
    slice_data,   # contains 2d slice of 3d regular mesh tally data
    origin='lower',
    norm=LogNorm(),
    extent=mesh.bounding_box.extent['xy']   # <- this PR enables getting the extent like this
) 
```

### CODE REVIEW

File: openmc/mesh.py [hunk 1]
```diff
@@ -793,6 +799,7 @@ def write_data_to_vtk(self, filename, datasets, volume_normalization=True):
             volume_normalization=volume_normalization
         )
 
+
```
Comment (shimwell) [inline openmc/mesh.py:802]:
```suggestion
```

File: tests/unit_tests/test_mesh.py [hunk 2]
```diff
@@ -33,3 +33,63 @@ def test_raises_error_when_flat(val_left, val_right):
     with pytest.raises(ValueError):
         mesh.upper_right = [25, 25, val_right]
         mesh.lower_left = [-25, -25, val_left]
+
```
Comment (shimwell) [inline tests/unit_tests/test_mesh.py:36]:
```suggestion


```

**Comment by shimwell** [comment]
Sorry for this mess of a PR, I accidentally merged a few branches together. I've tidied it up now. Great example of why squashing commits on merge is the way to go :smile:

File: tests/unit_tests/test_mesh.py [hunk 3]
```diff
@@ -33,3 +33,64 @@ def test_raises_error_when_flat(val_left, val_right):
...
+        np.array(
+            [
+                [ 1.,  2.],
+                [ 7.,  8.],
+                [13., 14.],
+                [19., 20.],
+                [25., 26.]
+            ]
+        )
+    )
```
Comment (shimwell) [inline tests/unit_tests/test_mesh.py:96]:
```suggestion
```

**Review by paulromano — APPROVED** [review]
Simple enough -- thanks @shimwell!

**Pull request merged**